CMakeLists.txt

```diff
@@ -86,6 +86,7 @@ set_option(TRACY_NO_CRASH_HANDLER "Disable crash handling" OFF)
 set_option(TRACY_TIMER_FALLBACK "Use lower resolution timers" OFF)
 set_option(TRACE_CLIENT_LIBUNWIND_BACKTRACE "Use libunwind backtracing where supported" OFF)
 set_option(TRACY_SYMBOL_OFFLINE_RESOLVE "Instead of full runtime symbol resolution, only resolve the image path and offset to enable offline symbol resolution" OFF)
+set_option(TRACY_ENABLE_IMAGE_CACHE "On glibc platforms, when doing offline symbol resolution, use an cache to determine the image path and offset instead of dladdr()" OFF)
```

Comment:
What is the reason for keeping this as an option?

Reply:
Hum, that is strange. I'll unoptimize the code and do some stepping again  in my case to make sure what I was seeing was actually correct (I'll hopefully get to it later today). But after the first symbol resolve was hit, I was missing symbol resolving for shared libraries loaded after. Looking through the libbacktrace code I could see it was using backtrace_state::fileline_fn as the control value to already called backtrace_initialize() (which was what was responsible for the shared library indexing), and that was set to something non-null after the call. That doesn't appear to be the case for you though, maybe I misinterpreted the results in the debugger with optimized code and something else was causing the issues.

Reply:
right, confirmed it's the same with HEAD client code. 
So I really only get elf_add() called a bunch of times, the first time it calls backtrace_initialize() and it indexes the .so with dl_iterate_phdr(). backtrace_initialize() get's called exactly once for me, on the first call to backtrace_pcinfo(), and I get many dynamic loaded of .so after that which are not included in the libbacktrace internal state as I nothing else called elf_add() for me after that.
In your case, is it not setting backtrace_state::fileline_fn or maybe something is resetting it?

Reply:
This is what I get with printf debugging:

```
backtrace_initialize: /proc/self/exe
elf_add: /proc/self/exe
elf_add: /proc/self/exe
elf_add: /usr/lib/libdebuginfod.so.1
elf_add:
elf_add: /usr/lib/libstdc++.so.6
elf_add: /usr/lib/libm.so.6
elf_add:
elf_add: /usr/lib/libgcc_s.so.1
elf_add: /usr/lib/libc.so.6
elf_add:
elf_add: /usr/lib/libcurl.so.4
elf_add:
elf_add: /usr/lib/libelf.so.1
elf_add:
elf_add: /lib64/ld-linux-x86-64.so.2
elf_add:
elf_add: /usr/lib/libnghttp2.so.14
elf_add:
elf_add: /usr/lib/libidn2.so.0
elf_add: /usr/lib/libssh2.so.1
elf_add:
elf_add: /usr/lib/libpsl.so.5
elf_add: /usr/lib/libssl.so.3
elf_add:
elf_add: /usr/lib/libcrypto.so.3
elf_add:
elf_add: /usr/lib/libgssapi_krb5.so.2
elf_add:
elf_add: /usr/lib/libzstd.so.1
elf_add:
elf_add: /usr/lib/libbrotlidec.so.1
elf_add:
elf_add: /usr/lib/libz.so.1
elf_add:
elf_add: /usr/lib/libunistring.so.5
elf_add: /usr/lib/libkrb5.so.3
elf_add:
elf_add: /usr/lib/libk5crypto.so.3
elf_add:
elf_add: /usr/lib/libcom_err.so.2
elf_add:
elf_add: /usr/lib/libkrb5support.so.0
elf_add:
elf_add: /usr/lib/libkeyutils.so.1
elf_add:
elf_add: /usr/lib/libresolv.so.2
elf_add:
elf_add: /usr/lib/libbrotlicommon.so.1
elf_add:
```

So as I understand it, the problem you have is that shared objects that are dlopened at a later time are not visible because libbacktrace has already collected a list of shared objects that the executable is linking to and will not look for more.

This feels like something libbacktrace should handle itself (in an ideal world). You already know when you want to reset libbacktrace state. Have you tried running `dl_iterate_phdr` to call `phdr_callback` instead?

Reply:
> So as I understand it, the problem you have is that shared objects that are dlopened at a later time are not visible because libbacktrace has already collected a list of shared objects that the executable is linking to and will not look for more

exactly.

> This feels like something libbacktrace should handle itself (in an ideal world).

I guess before I started looking more closely at the code I didn't realize you had already modified that libbacktrace code quite a bit and it wasn't just a dump from upstream. 

> you already know when you want to reset libbacktrace state. Have you tried running dl_iterate_phdr to call phdr_callback instead?

Yes, libacktrace code is already doing pretty much what I was doing once, so I can try to expose an update method I can call from outside - I'll try to do this this week and update this PR.

Reply:
> I guess before I started looking more closely at the code I didn't realize you had already modified that libbacktrace code quite a bit and it wasn't just a dump from upstream.

There have been some changes that add some sorely missing features, but there have been no substantial changes. The changes to `dl_iterate_phdr` are technical: 7e8961d2fc3

Reply:
> Yes, libacktrace code is already doing pretty much what I was doing once, so I can try to expose an update method I can call from outside - I'll try to do this this week and update this PR.

Please check if it would be feasible to implement this update functionality within libbacktrace itself.

Reply:
> The changes to dl_iterate_phdr are technical: https://github.com/wolfpld/tracy/commit/7e8961d2fc3a8883ba61738b201d04cddb15b795

I think this was needed because `elf_add` can now call debuginfod, which might issue network requests, and this didn't play well with the requirements of `dl_iterate_phdr`.

Reply:
>Please check if it would be feasible to implement this update functionality within libbacktrace itself.

I made an attempt at it in the latest commit